tests/ui/derives/lack-of-perfect-derives.stderr

```diff
@@ -0,0 +1,240 @@
+error[E0369]: binary operation `==` cannot be applied to type `S<X, X, X>`
+  --> $DIR/lack-of-perfect-derives.rs:23:15
+   |
+LL |     let _ = s == s2;
+   |             - ^^ -- S<X, X, X>
+   |             |
+   |             S<X, X, X>
+   |
+note: an implementation of `PartialEq` might be missing for `X`
+  --> $DIR/lack-of-perfect-derives.rs:15:1
+   |
+LL | struct X;
+   | ^^^^^^^^ must implement `PartialEq`
+help: consider annotating `X` with `#[derive(PartialEq)]`
+   |
+LL + #[derive(PartialEq)]
+LL | struct X;
+   |
+
+error[E0599]: the method `clone` exists for struct `S<X, X, Y>`, but its trait bounds were not satisfied
+  --> $DIR/lack-of-perfect-derives.rs:27:16
+   |
+LL | struct S<T, K, V>(Arc<T>, Rc<K>, Arc<V>, ());
+   | ----------------- method `clone` not found for this struct because it doesn't satisfy `S<X, X, Y>: Clone`
+...
+LL | struct X;
+   | -------- doesn't satisfy `X: Clone`
+...
+LL |     let s2 = s.clone();
+   |                ^^^^^ method cannot be called on `S<X, X, Y>` due to unsatisfied trait bounds
+   |
+note: trait bound `X: Clone` was not satisfied
+  --> $DIR/lack-of-perfect-derives.rs:4:10
+   |
+LL | #[derive(Clone, PartialEq)]
+   |          ^^^^^ unsatisfied trait bound introduced in this `derive` macro
+LL | struct S<T, K, V>(Arc<T>, Rc<K>, Arc<V>, ());
+   |          -  - implicit unsatisfied bound on this type parameter
+   |          |
+   |          implicit unsatisfied bound on this type parameter
+   = note: `#[derive(Clone)]` introduces `where`-bounds requiring every type parameter to implement `Clone`, even when not strictly necessary
+   = help: you can manually implement `Clone` with more targeted bounds: `impl<T, K, V> Clone for S<T, K, V> where V: Clone { /* .. */ }`
+help: consider annotating `X` with `#[derive(Clone)]`
+   |
+LL + #[derive(Clone)]
+LL | struct X;
+   |
```

Comment:
This is the case that "perfect derives" would fix, and main impetus behind this change. Note that the help still includes `where V: Clone`, because that obligation was met (because `V` ends up being `Y`, which is `Clone`. We could change this behavior slightly to not include that bound either, but ran out of time to look into that.

I would like to remove some of the output from this error, like the suggestion to make `X` cloneable, and maybe the secondary span label pointing at `S`, as that is already pointed at by the "unsatisfied trait bound `X: Clone`" note.